src/libraries/System.Data.OleDb/src/OleDbDataReader.cs

```diff
@@ -951,15 +951,15 @@ protected override DbDataReader GetDbDataReader(int ordinal)
             return GetData(ordinal);
         }
 
-        internal OleDbDataReader? ResetChapter(int bindingIndex, int index, RowBinding rowbinding, int valueOffset)
+        internal OleDbDataReader ResetChapter(int bindingIndex, int index, RowBinding rowbinding, int valueOffset)
         {
             return GetDataForReader(_metadata![bindingIndex + index].ordinal, rowbinding, valueOffset);
         }
 
-        private OleDbDataReader? GetDataForReader(IntPtr ordinal, RowBinding rowbinding, int valueOffset)
+        private OleDbDataReader GetDataForReader(IntPtr ordinal, RowBinding rowbinding, int valueOffset)
         {
             UnsafeNativeMethods.IRowsetInfo rowsetInfo = IRowsetInfo();
-            UnsafeNativeMethods.IRowset? result;
+            UnsafeNativeMethods.IRowset result;
```

Comment:
Do you have any sort of information on the native GetReferencedRowset below, or is this based more on the calling code assuming this returns non-null? Because the `null != result` below seems to indicate null as a possibility.

BTW if we're really 100% confident here, you can also remove the TODO below.

Reply:
Looking at the [docs](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ms721145(v=vs.85)) it can only be null when it's not a reference column in which case you will get hr=DB_E_NOTAREFERENCECOLUMN which would cause the excepton in ProcessResults. So to answer it can be null but probably easier to assume it's not as it's error state.

Reply:
I'll change it back to nullable and add Debug.Assert under Process result with a comment to make it clearer

Reply:
Thanks for checking that out, sounds good.

Reply:
I'll leave the null check though as there are some exceptional situations where ProcessResults will not throw (i.e. connection becomes null) - I suspect this is just for cleanup. I'll change it back to nullable but assume in the result is actually not null